[ADD] web_developer_mode_optional: Added module to allow access to devel...

alejandrosantana opened the pull request:
...oper mode, based on user groups.

Consider this as a substitute and improvement of PR https://github.com/OCA/web/pull/109

Now this allows to show "developer mode" link only to those users who are in a certain group.

~~**NOTE:**~~
- ~~To make the js work, I had to duplicate all code within `instance.web.Widget.extend({`, even though I only needed `on_menu_about`. But if I removed any of the rest, it didn't work as expected.~~
- ~~I have tried with _extend_ and _include_, with same results.~~

~~Have said that...~~
- ~~How do I properly extend a nameless function? (See chrome.js first line)~~

UPDATE: All strikethrough text above has been solved.

Commit:
[ADD] web_readonly_bypass: allow to save an onchange modifications on readonly fields

Commit:
[ADD] web_developer_mode_optional: Added module to allow access to developer mode, based on user groups.

Commit:
[FIX] web_developer_mode_optional: Fixes JS inheritance and removes unnecesary spaguetti code.

alejandrosantana commented:
OK, my problems with JS have been solved already. All spaguetti code have been removed.

Commit:
[FIX] web_developer_mode_optional: Fixed flake8 errors and some jslint.

anthony-muschang commented:
Tested :+1:

legalsylvain commented on web_developer_mode_optional/static/description/icon.svg:
```diff
@@ -0,0 +1,184 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Why did you add two version of the icon ? the svg file seems to me unnecessary ...

pedrobaeza replied:
Yeah, it should be only needed if you make any modifications, so that you allow to get the editable one available.

alejandrosantana replied:
@legalsylvain @pedrobaeza I just thought it could be useful for anyone that could modify the module to have the vectorial source of the image. But I can delete, if needed, of course. Should I take the "no .svg" as an OCA policy for any future module?

legalsylvain replied:
I can not confirm that it is an OCA policy, but I can say that it's the first time I see a vectorial icon in OCA modules. (But I don't know all of them)
@jgrandguillaume, @gurneyalex : What'is your PoV about that point ?

My _personal_ PoV is : 
- change icon is not a real need ;
- add svg file make PR on github and git diff less readable ;

But it's not a blocking point for me.

pedrobaeza replied:
You already have the source in https://github.com/OCA/maintainer-tools/blob/master/template/module/static/description/icon.svg, that's why I ask not to duplicate the file unless you make any modification.

legalsylvain replied:
@pedrobaeza It's not the same icon ! https://github.com/OCA/web/pull/128/files?short_path=e340032#diff-e34003280f7857fb9036ec1c6f49edc0

regards.

pedrobaeza replied:
Ah, sorry, browser doesn't show me the preview when I review it and I think it was the template one. Then I totally agree with Alejandro that the SVG must be there as en editable source. It's the same as if you include the compiled python pyc, but not the py :wink:

alejandrosantana replied:
@jgrandguillaume, @gurneyalex : Then, what do I proceed with OCA modules icons?
I think any module could have its own icon. In that case, for any future modification, it should include source (in this case an .svg file), even if it makes a commit less readable.
If no particular icon is used, then default to OCA .png-only icon, as it is on templates.

pedrobaeza replied:
@alejandrosantana, you have made it right. Include the source SVG when the icon is custom

alejandrosantana commented:
@jgrandguillaume, @gurneyalex Anything on this? If needed, I can remove the svg and dicuss its presence, yet I believe icon .svg source should be there for others to modify in the future (as I give permission to that).

pedrobaeza commented:
Alejandro, there's no problem with the SVG. The problem is having the time to review...

sbidoul committed:
[IMP] web_readonly_bypass: port to OCA from acsone-addons

Commit:
[CHG] some code review:
* Use a better test condition
* Fix the email address
[FIX]
* rename the path and web instance just as the module was

Commit:
cherry-pick 790a853 by pass readonly fields only if context is set with
readonly_by_pass. This context can either be a bool or field name array

Commit:
cherry-pick c639533 combine context, Get windows_action context in
BufferedDataSet Class

Commit:
fix readme as on v7

mga-odoo committed:
[ADD] New module <web_group_expand>

oihane committed:
[IMP] <web_group_expand> Make it work for v8
Adapt to OCA

Commit:
[UPD] web_developer_mode_optional: Updated version numbering to adapt to new OCA standard. Also set it as not application.

Commit:
[UPD] web_developer_mode_optional: Updated README file to adapt to latest OCA template. Updated icon to fit latest Odoo style.

Commit:
[FIX] web_developer_mode_optional: Minor fix in README file in title.

lmignon committed:
[FIX] Issue #193. display button as inline-block

hbrunn committed:
[FIX] double key in manifest dict

hparfr committed:
[FIX] web_environment_ribbon : Fix menu_navbar overlap

Set correct z-index on ribbon
Move position accordingly

Cf #191

sylvainc committed:
[IMP] web_environment_ribbon: conditional ribbon

sylvainc committed:
[IMP] Add default 'ribbon.name' configuration parameter

sylvainc committed:
[IMP] Update README.rst

sylvainc committed:
[REF] Use get_param() instead of search_read() to get the configuration parameter

sylvainc committed:
[IMP] Update roadmap section in README.rst

hparfr committed:
[IMP] web_environment_ribbon increase margin

+ 10px

sbidoul committed:
[ADD] addons table in README.md

hparfr committed:
[ADD] web_switch_company_warning : display a message saying it's time to reload the page

hparfr committed:
[fix] .rst formatting

pedrobaeza committed:
[IMP] web_group_expand: Add Jay Vora to credits

hparfr committed:
[FIX] .rst formating

hparfr committed:
[FIX] web_switch_company_warning

Remove place holders in readme
Add "use strict" in js

hparfr committed:
[FIX] web_switch_company_warning

Rename readme in README

hparfr committed:
[FIX] web_switch_company_warning

Fix travis build stuff.

hparfr committed:
[FIX] web_switch_company_warning

Missing , in __openerp_.py

hparfr committed:
[FIX] web_switch_company_warning

Fix typo in worker file name

Commit:
[CHG] Code review
* Improve .rst
* user_strict into js
* review Authors

Commit:
[CHG] Improve contributors into README.rst

sbidoul committed:
[UPD] addons table in README.md

sbidoul committed:
[UPD] addons table in README.md

sbidoul committed:
[UPD] prefix versions with 8.0

LeartS committed:
[web_tree_image] Add support resize option

set resize="<width>,<height>" in the field tag definition to
make the server return a resized version of the image.
Useful for saving bandwidth when dealing with large images and to
have uniform image sizes (aspect ratio is mantained by filling
emtpy areas with transparent background)

LeartS committed:
Version bump, update README

sbidoul committed:
[UPD] addons table in README.md

sbidoul committed:
[MOV] move addons out of __unported__ (they remain not installable)

bealdav committed:
[IMP] translation fr

alejandrosantana commented:
For some reason, OCA CLAbot is refusing my CLA, but I have signed, as I had contributed before.
Also: https://odoo-community.org/members?name=Santana

pedrobaeza commented:
Not exactly, Alejandro. It's talking about other comits in your PR. You have made the rebase in a strange way that have mixed older commits (you count 43 in this PR). Please fix this.

alejandrosantana commented:
OK, time to learn a bit more git.

alejandrosantana commented:
Just gone the easy way: make a clean fork, new branch and new clean PR in #248 
Sorry for inconveniences, but this way is way cleaner, I think.
